runtime/v2/vm/src/main/java/com/walmartlabs/concord/svm/Frame.java

```diff
@@ -109,6 +113,19 @@ public Serializable getLocal(String k) {
         return Collections.unmodifiableMap(locals);
     }
 
+    public <T> T withLock(Callable<T> c) {
```

Comment:
I think you can achieve the same by using `synchronized` on this method instead of a Lock object.

Reply:
yep